universal-login-relayer/lib/services/IdentityService.js

```diff
@@ -27,6 +29,12 @@ class IdentityService {
 
   async executeSigned(contractAddress, message) {
     const contract = new ethers.Contract(contractAddress, this.abi, this.wallet);
+    const addKeySighash = new Interface(Identity.interface).functions.addKey.sighash;
+    if (message.to === contractAddress && message.data.slice(0, addKeySighash.length) === addKeySighash) {
+      const [address] = (this.codec.decode(['bytes32', 'uint256', 'uint256'], message.data.replace(addKeySighash.slice(2), '')));
+      const key = utils.hexlify(utils.stripZeros(address));
```

Comment:
Two lines above could be easily extracted to function, eg `const [key, address] = decodeData(addKeySighash);`